appinventor/appengine/src/com/google/appinventor/client/editor/simple/components/MockChart.java

```diff
@@ -80,4 +91,15 @@ public void onPropertyChange(String propertyName, String newValue) {
             setBackgroundColorProperty(newValue);
         }
     }
+
+    @Override
+    protected boolean acceptableSource(DragSource source) {
+        MockComponent component = null;
+        if (source instanceof MockComponent) {
+            component = (MockComponent) source;
+        } else if (source instanceof SimplePaletteItem) {
+            component = (MockComponent) source.getDragWidget();
+        }
+        return component instanceof MockLineChartData;
```

Comment:
This should be changed in the future (either abstract this method away, or allow all types)

Reply:
Also , for future considerations, for Charts that support multiple Data Component types, some filtering could be done.

E.G. Bar Chart could support key-value data and coordinate data. In that case, if coordinate data already exists, key-value data could be disallowed.